pkg/kubelet/util/podConverter/podConverter.go

```diff
@@ -134,6 +139,18 @@ func ConvertVmPodToContainerPod(pod *v1.Pod) *v1.Pod {
 	return cpod
 }
 
+// Libvirt guestOS cpu resource doc:
+// https://libvirt.org/docs/libvirt-appdev-guide-python/en-US/html/libvirt_application_development_guide_using_python-Guest_Domains-Domain_Config-Memory_CPU.html
+// Libvirt VM domain vcpus specifies the limit of number of vcpus the guest OS can have
+// Arktos vm runtime doesn't support guest OS vcpu reset currently, the VM workload should be classified as Guaranteed
+// resource model.
+// API validation should enforce this when accepting the VM POD to avoid resource level misalignment since scheduler uses
+// requested resources in its Predicate function
+//
+func getVCpuCountForVm(vmSpec * v1.VirtualMachine) int64 {
+	return vmSpec.Resources.Limits.Cpu().Value()
```

Comment:
Can we also handle the situation that limit is not specified but request is specified?

Reply:
this should be in the admission controller for VM type as i mentioned in the PR which is not include in this PR as the original thought. i.e. the admission controller should enforce the limits is set and if request is also set, it should be equal to the limitvalue ( give only one vm is in the pod in our current design ). let me add those part as well if this is agreed to the right solution here as described below:

we can expand this PR with :

1. this current bug fix to set up the cpu count for VM --- which is in this commit.
2. admission controller to enforce the limit setting of VM, and if request is also set, it should be same as limit -- which is Guaranteed QoS class for VM type
3. fix bug in QoS to add VM type check as well.

Reply:
I see. I think your code comment can simply say "for now we only support Guaranteed QoS class for VM pods".  It will clearly explain why you only check .limit.

Reply:
Also: if you plan to do this way, you should check if .limit is available before read it, just in case the admission controller is disabled.

Reply:
that should be OK. it just returned 0 as i tested it.